Project Complete: feature/structure

Elbie-em opened the pull request:
Please follow the map below to see what we worked on. The requirements for this project can be found [here](https://www.theodinproject.com/courses/javascript/lessons/tic-tac-toe-javascript).
* Set up project with an HTML, CSS and JavaScript files
* Added Linter configurations and files for JavaScript(eslint) and CSS(style lint)
* Built functions that allow players to add marks to a specific cell on the board, and then tie it to the DOM, letting players click on the game board to place their marker.
* Built logic that checks when the game is over by declaring a winner or tie.
* Cleaned up the interface to allow players to put in their names, included a button to start/restart the game and add a display element that congratulates the winning player!
* Added validation in input form.

Elbie-em committed:
Intialize npm

Elbie-em committed:
Add gitignore file

Elbie-em committed:
install npm eslint

Elbie-em committed:
Add eslint config file

Elbie-em committed:
install npm stylelint

Elbie-em committed:
Add stylelint config file

Elbie-em committed:
add HTML5 structure

Elbie-em committed:
add game headings and containers

Elbie-em committed:
link styles.css to index

Elbie-em committed:
add styling for game board cell

Elbie-em committed:
add form for player information

Elbie-em committed:
start build on gamelogic

abdelp committed:
Add gameInit method

- instantiate players
- instantiate game
- hide players form
- enable board
- display user turn

abdelp committed:
Add posible structure for game

Elbie-em committed:
Add possibility of cell selection and array communication

Elbie-em committed:
add logic that keeps players from playing in spots that are already taken

Elbie-em committed:
add logic that keeps board hidden before game starts

abdelp committed:
Add checkWinner method

abdelp committed:
Remove spaces

Elbie-em committed:
fixed cell disorientation

Elbie-em committed:
add restart button

Elbie-em committed:
refactor code to meet requirements

abdelp committed:
Merge branch 'feature/structure' of https://github.com/Elbie-em/JavaScript-Tic-Tac-Toe-Game into feature/structure

abdelp committed:
Add README.md content and app screenshot

abdelp committed:
Fix automatically eslint offenses

abdelp committed:
Add possible structure for moludes

abdelp committed:
Add user names check

Elbie-em committed:
add display board method and fix linter errors

Elbie-em committed:
Update live demo link in README file

Elbie-em committed:
Update restart button

maelfosso commented on js/modules/game.js:
```diff
@@ -0,0 +1,97 @@
+  };
+};
+
+export { Game }; // eslint-disable-line import/prefer-default-export
```
Please, re-enable that configuration and apply a default export.

maelfosso commented on js/index.js:
```diff
@@ -0,0 +1,79 @@
+import { Game } from './modules/game.js'; // eslint-disable-line import/extensions
+
+const Player = (id, name, symbol) => ({ id, name, symbol });
```
Move it into `game.js`

maelfosso replied:
Make the `game.js` file the module with no code related to DOM.
I mean code with only Logic, controller, etc...
And into `index.js` only DOM manipulation in respect to the output of `game.js`

maelfosso reviewed (changes requested):
# Changes required :red_circle: 

Hello @Elbie-em @abdelp 

You did a really nice job.
Nothing really to say except some small changes to do.

Kindly read the comments left below and apply them. Then, submit a new code review request.
Thanks

Mael FOSSO

Elbie-em committed:
add doman.js

Elbie-em committed:
refactor game.js

abdelp committed:
Update dom manipulation

Remove start button onclick event from index.js
Remove displayBoard from game.js
Remove hidePlayerTurn from game.js
Add to doman.js module

abdelp committed:
Add checkPlayersName function

abdelp committed:
Update message display

Remove displayPlayerTurn from game.js to doman.js
Add checkPlayersNames method
Add displayError method

abdelp committed:
Add possible estructure for selectCell

abdelp committed:
Almost finished

abdelp committed:
Almost finished

abdelp committed:
Add changes

abdelp committed:
Update checkWinner method to check diagonals correctly

abdelp committed:
Remove return from selectCell method

abdelp committed:
Add mjs files for modules

abdelp committed:
Add missing .js for game module

abdelp commented:
Fixed all of the issues moving the DOM manipulation to the doman.js module

whiz25 reviewed (approved):
## Approved! :clap: 
You have the requested changes and it's a :+1:  from me. Merge and proceed to next milestone.

### Martin Afani

abdelp committed:
Fix display current user when select a cell
